src/test/java/com/cdancy/jenkins/rest/features/JobsApiLiveTest.java

```diff
@@ -114,123 +115,223 @@ public void testGetBuildInfo() {
+        boolean success = api().disable("job/test-folder", "JobInFolder");
+        assertTrue(success);
+    }
+
+    @Test(dependsOnMethods = "testDisableJobInFolder")
+    public void testEnableJobInFolder() {
+        boolean success = api().enable("job/test-folder", "JobInFolder");
+        assertTrue(success);
+    }
+
+    @Test(dependsOnMethods = "testEnableJobInFolder")
+    public void testSetDescriptionOfJobInFolder() {
+        boolean success = api().description("job/test-folder", "JobInFolder", "RandomDescription");
+        assertTrue(success);
+    }
+
+    @Test(dependsOnMethods = "testSetDescriptionOfJobInFolder")
+    public void testGetDescriptionOfJobInFolder() {
+        String output = api().description("job/test-folder", "JobInFolder");
+        assertTrue(output.equals("RandomDescription"));
+    }
+
+    @Test(dependsOnMethods = "testGetDescriptionOfJobInFolder")
+    public void testGetJobInfoInFolder() {
+        JobInfo output = api().jobInfo("job/test-folder", "JobInFolder");
+        assertNotNull(output);
+        assertTrue(output.name().equals("JobInFolder"));
+        assertTrue(output.builds().isEmpty());
+    }
+
+    @Test(dependsOnMethods = "testGetJobInfoInFolder")
+    public void testBuildWithParameters() {
+        Map<String, List<String>> params = new HashMap<>();
+        params.put("SomeKey", Lists.newArrayList("SomeVeryNewValue"));
+        queueIdForAnotherJob = api().buildWithParameters("job/test-folder", "JobInFolder", params);
+        assertNotNull(queueIdForAnotherJob);
+        assertTrue(queueIdForAnotherJob > 0);
+    }
+
+    @Test(dependsOnMethods = "testBuildWithParameters")
+    public void testLastBuildTimestampOfJobInFolder() {
+        String output = api().lastBuildTimestamp("job/test-folder", "JobInFolder");
+        assertNotNull(output);
+    }
+
+    @Test(dependsOnMethods = "testLastBuildTimestampOfJobInFolder")
+    public void testGetProgressiveText() {
+        ProgressiveText output = api().progressiveText("job/test-folder", "JobInFolder", 0);
+        assertNotNull(output);
+        assertTrue(output.size() > 0);
+        assertFalse(output.hasMoreData());
+    }
+
+    @Test(dependsOnMethods = "testGetProgressiveText")
+    public void testGetBuildInfoOfJobInFolder() {
+        BuildInfo output = api().buildInfo("job/test-folder", "JobInFolder", 1);
+        assertNotNull(output);
+        assertTrue(output.fullDisplayName().contains("JobInFolder #1"));
+        assertTrue(output.queueId() == queueIdForAnotherJob);
+    }
+
+    @Test(dependsOnMethods = "testGetBuildInfoOfJobInFolder")
+    public void testDeleteJobInFolder() {
+        RequestStatus success = api().delete("job/test-folder", "JobInFolder");
+        assertTrue(success.value());
+    }
+
+    @Test(dependsOnMethods = "testDeleteJobInFolder")
+    public void testDeleteFolder() {
+        RequestStatus success = api().delete(null, "test-folder");
+        assertTrue(success.value());
+    }
```

Comment:
Is there a way to tell programmatically if the plugin is installed so that we can potentially skip these tests if it isn't?

Reply:
Maybe as part of a first test in this chain of tests check if we can create a folder and if we can't then skip the rest of the tests?

Reply:
Agreed. I actually think this is a requirement before we push forward with this PR. I'll see if I can knock something out quick unless others want to do so.

Reply:
Ah.. I like this idea to check if the plugin exists before we write the test.

Reply:
Got something going now and will push to master shortly. You can then rebase off of that and check the list of plugins installed for what you need.

Reply:
We could prevalidate a job's config before we post it to jenkins using this -->  http://localhost:8080/pluginManager/prevalidateConfig API. It returns back the list of plugins that are not installed or outdated. Check [This](https://issues.jenkins-ci.org/browse/JENKINS-15003?focusedCommentId=167568&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-167568) link out.

Reply:
So we could check if it's present, and if not, then attempt to install it. Very awesome.

Reply:
PullRequest for PluginManagerApi put together [HERE](https://github.com/cdancy/jenkins-rest/pull/17). Have a look guys and let me know what you think. It only supports the listing of the plugins which _should_ be enough to get us around this issue and then give us something to build upon.

Reply:
@Dhivyaa21 I've merged that PR to master if you want to rebase off of that and use the new `PluginManagerApi.plugins()` call to test if a given plugin is available.

Reply:
@cdancy - sure!